src/java/org/apache/nutch/api/NutchServer.java

```diff
@@ -108,8 +118,41 @@ public NutchServer() {
     application.setStatusService(new ErrorStatusService());
     childContext.getAttributes().put(NUTCH_SERVER, this);
 
-    // Attach the application.
-    component.getDefaultHost().attach(application);
+    AuthenticationTypeEnum authenticationType = configManager.get(ConfigResource.DEFAULT)
+            .getEnum("restapi.auth", AuthenticationTypeEnum.NONE);
+
+    if (authenticationType == AuthenticationTypeEnum.NONE) {
```

Comment:
Please change to the more efficient switch notation. Multiple If's are messy and JDK1.8 has better switch support for string input values.